File: mgi/src/controllers/IControllers/IAuthController.ts
import { Request, Response, NextFunction } from 'express';

export default interface IAuthController {
    session(req: Request, res: Response, next: NextFunction);
    login(req: Request, res: Response, next: NextFunction);
    signup(req: Request, res: Response, next: NextFunction);
    logout(req: Request, res: Response, next: NextFunction);
}

File: mpt/Controllers/IControllers/IAuthController.cs

using Microsoft.AspNetCore.Mvc;
using Mpt.Dtos;

namespace Mpt.IControllers
{
    public interface IAuthController
    {

        Task<ActionResult<UserWithRoleDto>> Login(UserLoginDto login);
        Task<ActionResult<UserWithRoleDto>> Signup(SignupUserDto user);
        Task<ActionResult<UserWithRoleDto>> Session();
        Task<ActionResult<UserWithRoleDto>> ValidateToken(TokenDto token);
        ActionResult Logout();


    }
}